torchvision/datasets/mnist.py

```diff
@@ -418,7 +419,8 @@ def download(self) -> None:
     def __getitem__(self, index: int) -> Tuple[Any, Any]:
         # redefined to handle the compat flag
         img, target = self.data[index], self.targets[index]
-        img = Image.fromarray(img.numpy(), mode='L')
+        if self.get_image_backend() == 'PIL':
+            img = Image.fromarray(img.numpy(), mode='L')
```

Comment:
Same comment here about the data ordering and ndarray. Should we be consistent and always return `CHW` format, with C potentially being 1?